TCAT/Controllers/ServiceAlertsViewController.swift

```diff
@@ -0,0 +1,175 @@
+            } else {
+                make.top.equalToSuperview().offset(view.layoutMargins.top)
+            }
+        }
+    }
+    
+    func getServiceAlerts() {
+        Network.getAlerts().perform(withSuccess: { (request) in
+            if (request.success) {
+                self.alerts = self.sortedAlerts(alertsList: request.data)
+            }
+        }) { (error) in
+            let fileName = "ServiceAlertsVieController"
+            let line = "\(fileName) \(#function): \(error)"
+            print(line)
+        }
+    }
+    
+    func sortedAlerts(alertsList: [Alert]) -> [Int: [Alert]] {
+        var sortedAlerts = [Int: [Alert]]()
+        for alert in alertsList {
+            if var alertsAtPriority = sortedAlerts[alert.priority] {
+                alertsAtPriority.append(alert)
+                sortedAlerts[alert.priority] = alertsAtPriority
+            } else {
+                priorities.append(alert.priority)
+                sortedAlerts[alert.priority] = [alert]
+            }
+        }
+        priorities.sort()
+        
+        return sortedAlerts
+    }
+}
+
+extension ServiceAlertsViewController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = HeaderView()
+        
+        switch priorities[section] {
+        case 0:
+            headerView.setupView(labelText: Constants.TableHeaders.highPriority)
+        case 1:
+            headerView.setupView(labelText: Constants.TableHeaders.mediumPriority)
+        case 2:
+            headerView.setupView(labelText: Constants.TableHeaders.lowPriority)
+        default:
+            headerView.setupView(labelText: Constants.TableHeaders.noPriority)
+        }
+        
+        return headerView
+    }
+}
+
+extension ServiceAlertsViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return priorities.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if let alerts = alerts[priorities[section]] {
+            return alerts.count
+        } else {
+            return 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: ServiceAlertTableViewCell.identifier, for: indexPath) as? ServiceAlertTableViewCell
+        
+        if let alertList = alerts[priorities[indexPath.section]] {
+            cell?.alert = alertList[indexPath.row]
+            cell?.rowNum = indexPath.row
+            cell?.setData()
+            cell?.setNeedsUpdateConstraints()
+        }
+        
+        return cell!
```

Comment:
Hmm, not a fan of `!`. Is there a version of `dequeueReusable` that isn't optional?